rust/alloc/slice.rs

```diff
@@ -496,6 +514,36 @@ impl<T> [T] {
         hack::to_vec(self, alloc)
     }
 
+    /// Tries to copy `self` into a new `Vec` with an allocator.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(allocator_api)]
+    ///
+    /// use std::alloc::System;
+    ///
+    /// let s = [10, 40, 30];
+    /// let x = s.try_to_vec_in(System).unwrap();
+    /// // Here, `s` and `x` can be modified independently.
+    /// ```
+    #[inline]
+    #[stable(feature = "kernel", since = "1.0.0")]
+    pub fn try_to_vec_in<A: Allocator>(&self, alloc: A) -> Result<Vec<T, A>, TryReserveError>
+    where
+        T: Clone,
+    {
+        let mut v = Vec::try_with_capacity_in(self.len(), alloc)?;
+        // SAFETY:
+        // allocated above with the capacity of `self`, and initialize to `self.len()` in
+        // ptr::copy_to_non_overlapping below.
+        unsafe {
+            self.as_ptr().copy_to_nonoverlapping(v.as_mut_ptr(), self.len());
```

Comment:
This is clearly wrong. `T` only implements `Clone` and you cannot just do a bitwise copy. The best approach is to add a `try_to_vec` method to trait `hack::ConvertVec`. (You'll need to move the gate from trait `ConvertVec` to its method).

Reply:
D'oh, yes, I copied the `Copy` one instead of the `Clone` one from `ConvertVec`. My bad.